src/base/memory_monitor_sr.cpp

```diff
@@ -402,19 +479,69 @@ namespace cubperf
     m_mmon = mmon;
   }
 
-  int memory_monitor::aggregater::get_server_info (const MMON_SERVER_INFO &info)
+  void memory_monitor::aggregater::get_server_info (MMON_SERVER_INFO &info)
   {
-    return 0;
+    strcpy (info.name, m_mmon->m_server_name.c_str ());
+    info.total_mem_usage = m_mmon->m_total_mem_usage.load ();
   }
 
-  int memory_monitor::aggregater::get_module_info (const MMON_MODULE_INFO &info, int module_index)
+  int memory_monitor::aggregater::get_module_info (MMON_MODULE_INFO &info, int module_index)
   {
-    return 0;
+    int error = NO_ERROR;
+
+    error = m_mmon->m_module[module_index]->aggregate_stats (info);
+    return error;
   }
 
-  int memory_monitor::aggregater::get_transaction_info (const MMON_TRAN_INFO &info, int tran_count)
+  int memory_monitor::aggregater::get_transaction_info (MMON_TRAN_INFO &info, int tran_count)
   {
-    return 0;
+    int num_total_indices = 0;
+    LOG_TDES **trantable;
+    LOG_TDES *tdes = nullptr;
+    std::vector <std::pair <int, uint64_t>> tran_info;
+
+    trantable = logtb_get_trantable_nolatch (&num_total_indices);
+
+    for (int i = 0; i < num_total_indices; i++)
+      {
+	tdes = trantable[i];
+	if (tdes != NULL && tdes->trid != NULL_TRANID && tdes->state == TRAN_ACTIVE)
+	  {
+	    tran_info.push_back (std::make_pair (tdes->trid, tdes->cur_mem_usage.load ()));
+	  }
+      }
+
+    if (tran_count == 0)
+      {
+	info.num_tran = tran_info.size ();
+      }
+    else
+      {
+	if (tran_count > tran_info.size ())
+	  {
+	    info.num_tran = tran_info.size ();
+	  }
+	else
+	  {
+	    info.num_tran = tran_count;
+	  }
+	qsort (&tran_info.front (), tran_info.size (), sizeof (tran_info[0]), comp);
```

Comment:
I suggest std::sort, and to define comp() only in this context.
```suggestion
        std::sort (tran_info.begin (), tran_info.end (), [] (const auto& tran_pair1, const auto& tran_pair2)
            {
              return tran_pair1.second > tran_pair2.second; // mem usage comparison
            });
```

Reply:
or 
```
        const auto& comp = [] (const auto& tran_pair1, const auto& tran_pair2)
        {
          return tran_pair1.second > tran_pair2.second; // mem usage comparison
        };
        std::sort (tran_info.begin (), tran_info.end (), comp);
```

Reply:
I was not proficient in using C++ STL and was unaware of better approaches. Thank you for the advice.